src/compiler/sys.ts

```diff
@@ -16,7 +16,7 @@ namespace ts {
         createDirectory(path: string): void;
         getExecutingFilePath(): string;
         getCurrentDirectory(): string;
-        readDirectory(path: string, extension?: string, exclude?: string[]): string[];
+        readDirectory(path: string, extension?: string, exclude?: string[], depth?: number): string[];

```

Comment:
Does the ChakraHost.readDirectory implementation honor this new parameter you added? For ChakraHost we simply delegate to the host implementation or readDirectory.